mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/MailboxMapper.java

```diff
@@ -44,6 +44,11 @@
      */
     MailboxId create(Mailbox mailbox) throws MailboxException;
 
+    /**
+     * Create a {@link Mailbox} with the given {@link MailboxPath} and uid to the underlying storage
+     */
+    Mailbox create(MailboxPath mailboxPath, long uidValidity) throws MailboxException;
```

Comment:
I would be picky, I would ask for UidValidity strong typing ;-)

Reply:
Was not the primary goal but while we are at it, sure :)

Reply:
In an other PR though

Reply:
#3132